Minor UI updates

bonomat opened the pull request:
Commits are not perfect unfortunately. 

![image](https://user-images.githubusercontent.com/224613/156944306-fd302500-e018-4d2f-adb9-fd2ea1afb0ee.png)

TODO: fix funding rate. The funding rate in the nav bar might not yet be offer dependent.

bonomat committed:
Hide disclaimer checkbox

If checked, the disclaimer won't be shown again

bonomat committed:
Move connection status outside to remove duplication

bonomat committed:
Remove unused variable

bonomat committed:
No maker alert! We want toast

Also fixes alertbox bug:
The problem of the appearing and disappearing alert box is that `connectedToMakerOrUndefined` is undefined if you are not connected to your daemon or if we have not yet received the event. We set `connectToMaker` to `offline: false` and think the maker is offline. Once we get a connection to our daemon, and get the information about the maker connection, we set `connectToMaker` to `online: true`.

For a short time after a page refresh we might not yet have received an event from our daemon and hence believe we are not connected to the maker.

bonomat committed:
Fix length of closed positions accordion

Also remove unneded vstack

bonomat committed:
Make connection to daemon error disappear again if connection is back

bonomat committed:
Add padding to position to not stick to button

bonomat committed:
Reduce duplication by making use of layout route

da-kami commented on taker-frontend/src/App.tsx:
```diff
@@ -157,49 +193,46 @@ export const App = () => {
+                                        <NavigationButtons/>
                                         <Trade
                                             longOrder={longOrder}
                                             shortOrder={shortOrder}
```
Why would we pass long and short order into the `Trade` component if that component is now either long or short? I think it should just be order now...

bonomat replied:
In this commit we do not have yet have the split.

da-kami commented on taker-frontend/src/App.tsx:
```diff
@@ -159,7 +156,7 @@ export const App = () => {
     useEffect(() => {
         const id = "maker-connection-toast";
-        if (!connectedToMaker.online && !toast.isActive(id)) {
+        if (connectedToMakerOrUndefined && !connectedToMakerOrUndefined.online && !toast.isActive(id)) {
```
Doesn't this mean, if we start the application and never connect to a maker, that our `connectedToMaker` status is set to `{ online: false }` but we never display the toast?

Is that wanted behavior?

bonomat replied:
yes, because otherwise we get this flickering behavior: 
1. start: `connectedToMakerOrUndefined = undefined `
2. loaded: connectedToMakerOrUndefined = online = true`

da-kami replied:
I think that was expected (and wanted) behavior. Otherwise you would not see the toast if the maker is offline and you open the UI.

Either way, I think given that we have the the maker's status in color on the top, and we show skeletons for price, etc. if we don't have an offer it might not be necessary to show the toast if we *initially* don't have a connection but only show the toast upon losing the connection.

@klochowicz any opinion on that?

bonomat replied:
> I think that was expected (and wanted) behavior. Otherwise you would not see the toast if the maker is offline and you open the UI.

that's wrong. You will see the toast if you are online and the maker is offline. But you won't see the toast if you are offline. 

you're online meaning: your daemon is running or and you have seen the event. 
If you are not online (your daemon is not running) or have not seen the event yet, you can't reason about the maker being here or not.

bonomat replied:
speaking about this: the same is wrong with the online status in the navbar :)

da-kami commented on taker-frontend/src/App.tsx:
```diff
@@ -182,7 +182,7 @@ export const App = () => {
 
         {closedPositions.length > 0
-            && <Accordion allowToggle width={"100%"}>
+            && <Accordion allowToggle width={"80%"}>
```
I am not sure I can follow why this is better, but I don't mind much.

bonomat replied:
This is just a small UI change making the accordion not using 100% of the width.

da-kami reviewed (changes requested):
Thanks for tackling the UI issues and adding improvements. 

---

Some issues I found:

Should be investigated and fixed within this PR:
- It seems the confirmation screen is not correctly initialized, see my comment above

Can be done in a follow-up, but is necessary to ship the taker short feature:

- Given that we have two separate funding rates now, the funding rate has to be applied per order and cannot be part of the global trading params anymore but part of the order. We will likely have to display it inside `Trade` and not in the top-bar.

---

Some overall feedback: 

The commits are not atomic and it was confusing to review this by commit. 
There are quite a lot of changes in commits that are not part of what the commit message describes. Better squash then, otherwise it's just harder to review.

Additionally, some things grew on the way, but are in separate commits. Especially the change from long-/short-order to just order in `Trade` should have been one commit. I wrote down that this should be cleaned up and only noticed later that it was. Better squash this next time, or create a cleaner commit history, otherwise reviewing is a pain.

bonomat committed:
Make toast

By changing the alert box to a toast we avoid jumpy components.
Toasts are overlays and don't move things around.

bonomat committed:
Introduce long and short navigation button

bonomat committed:
Make trade component long/short configurable

da-kami commented on taker-frontend/src/components/ConfirmOrderModal.tsx:
```diff
@@ -69,9 +73,10 @@ export default function ConfirmOrderModal({
                     <HStack>
                         <Text>
-                            Market sell <b>{quantity}</b> of BTC/USD @
+                            Market {buy_or_sell}
```
```suggestion
                            Market {buy_or_sell}&nbsp;
```

I think the linter / dprint removed that space for you - probably better to add an explicit one with `&nbsp;` otherwise it reads like this:

![image](https://user-images.githubusercontent.com/5557790/157171260-0276a1ba-64af-4999-a495-e5f495d373ce.png)

bonomat replied:
Good catch

da-kami reviewed (approved):
Thanks!
